source/extensions/compression/gzip/compressor/BUILD

```diff
@@ -26,6 +26,9 @@ envoy_cc_extension(
     name = "config",
     srcs = ["config.cc"],
     hdrs = ["config.h"],
+    extra_visibility = [
```

Comment:
Unfortunately the reason this wasn't visible is we don't allow Envoy core code to depend on extensions.

Reply:
just for the reference (mostly for myself so it would be easier to find in future). this should be reworked because of https://github.com/envoyproxy/envoy/issues/9953

Reply:
Yeah, it's a relatively new policy.  Before we occasionally broke people's builds if they opted out of certain extensions, then fixed it after they called it out.  Better to not break in the first place :-P

If there's a place we could call that out where you would have seen it, please improve the docs in this area.  It's totally intuitive if something fails to build due to visibility to just fix it.  Normally we put such checks in the format scripts, but unfortunately we can't with this one until we clean up all the legacy visibility rules.

Reply:
changes visibility to test/extensions. i hope this is ok (for test/extensions to depends on source/extensions)?

Reply:
why is it needed? I think we also don't want cross-extension dependencies, so explicit lists would be better.  If that's onerous we could at least say //test/extensions/filters/http:__subpackages__ ? I'll defer to @lizan if he has thoughts/preferences.

Reply:
yeah. looks like i still can use this target in tests even w/o explicit "extra_visibility". will remove